### ISSUE TITLE
Memory leak in Tribler

### ISSUE THREAD

**Issue by whirm** [Tribler/tribler#847]
I was taking a look at the automated week long run and I saw that we are leaking sets of CallLater+PyTimer:

```
Total 2978723 objects, 741 types, Total size = 860.9MiB (902748432 bytes)
 Index   Count   %      Size   % Cum     Max Kind
     0  407301  13 456177120  50  50    1120 CallLater
     1  407300  13 140111200  15  66     344 PyTimer
     2  437030  14 134613776  14  80  786712 dict
     3  883489  29  70679120   7  88      80 instancemethod
     4  409065  13  19635120   2  90      48 PySwigObject
     5   53671   1  12275536   1  92 6872000 list
     6  138502   4  11641271   1  93  202477 str
     7    1380   0   4714080   0  94    3416 SwiftDownloadImpl
     8    3538   0   3934256   0  94    1112 Message
     9     721   0   3384912   0  94  196944 module
    10    4334   0   3002280   0  95  196904 OrderedDict
    11   23511   0   2821320   0  95     120 function
    12   20249   0   2591872   0  95     128 code
    13   34160   1   2583768   0  96   80056 tuple
    14    2224   0   2473088   0  96    1112 FullSyncDistribution
    15    2069   0   1870376   0  96     904 type
    16       2   0   1573456   0  96  786728 LimitedOrderedDict
    17    1385   0   1550816   0  96    1120 CallbackConfigParser
    18    1382   0   1547840   0  97    1120 SwiftDef
    19    1312   0   1458944   0  97    1112 Member

In [21]: pt = om.get_all("PyTimer")

In [22]: pt[0].p
Out[22]: 
[list(140390402911768 6872000B 814596refs 2par),
 CallLater(140390029254024 1120B 19refs 2par)]

In [23]: pt[0].p[0].p
Out[23]: 
[dict(140390380897456 1816B 48refs 1par),
 module(140390380917768 3408B 48refs 'gc')]

In [24]: pt[0].p[1].p
Out[24]: 
[list(140390402911768 6872000B 814596refs 2par),
 instancemethod(140390047573488 80B 3refs 1par)]

In [25]: pt[0].p[0].p[0].p
Out[25]: [dict(140390403130824 3352B 142refs)]

In [26]: pt[0].p[0].p[1].p
Out[26]: []

In [27]: pt[0].p[1].p[0].p
Out[27]: 
[dict(140390380897456 1816B 48refs 1par),
 module(140390380917768 3408B 48refs 'gc')]

In [28]: pt[0].p[1].p[1].p
Out[28]: [PyTimer(140390048600784 344B 5refs 2par)]
```

For the full run and memory dumps, look at: 
http://jenkins.tribler.org/job/Test_performance_Tribler_idle_1week_nightly_devel/9/

**Comment by whirm** [comment]
Debian doesn't have this patched yet, maybe we will have to do some monkeypatching.

**Comment by whirm** [comment]
@Baigle I guess we will patch our windows/OSX systems and do some monkeypatching for linux.

**Comment by devos50** [comment]
@whirm what's the status of this issue? From the url @egbertbouman posted, it seems to be fixed.

**Comment by whirm** [comment]
Yes, that got fixed in wx 2.8.12, and we aren't even using WX2.8 that anymore :D
